app/Community/Actions/LoadThinActivePlayersListAction.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Community\Actions;
+
+use App\Enums\Permissions;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Cache;
+
+class LoadThinActivePlayersListAction
+{
+    /**
+     * @return array<array{
+     *   user_id: int,
+     *   game_id: int,
+     *   username: string,
+     *   display_name: string,
+     *   rich_presence: string,
+     *   game_title: string,
+     * }>
+     */
+    public function execute(): array
+    {
+        return Cache::flexible('all-active-players', [20, 60], function () {
```

Comment:
Data is considered fresh 20 seconds after it's fetched.

After 20 seconds of the data being fresh, but sooner than 60 seconds, the server will retry to update the data in the background.

If no requests come in after 60 seconds, the data is considered stale and the cache will be purged.

With this approach, realistically on the home page there will never be an unlucky user who gets a stale cache hit.